### PR TITLE
[controller][admin-tool][test] Wire nearline write throughput quota through controller SOT

### PR DESCRIPTION
**Pull request by misyel**
## Problem Statement

The nearline **write quota** (the rate at which a store may produce real-time
data) is enforced externally but the Venice controller
has no authoritative record of it. The store-level fields that model this —
`throughputQuotaInBytes` and `throughputQuotaInRecords` — were reserved in the
`AdminOperation` (v101) and `StoreMetaValue` (v46) Avro schemas but were left
completely unwired: they could not be set, persisted, or read back through the
controller. As a result the controller cannot serve as the source of truth for
the write quota, and there is no supported way to configure it per store.

## Solution

Wire the two existing store-level fields end-to-end so the controller becomes
the source of truth for the nearline write quota. **No new schema version is
required** — the fields already exist in the current-latest schemas; the default
of `-1` continues to mean "no limit".

The wiring mirrors the proven `maxNearlineRecordSizeBytes` / `storageQuotaInByte`
patterns exactly:

- **Store model** (`Store`, `ZKStore`, `ReadOnlyStore`, `SystemStore`, `StoreInfo`):
  getters/setters backed by the existing `StoreProperties` Avro fields; read-only
  variants throw `UnsupportedOperationException`; `StoreInfo.fromStore` mapping.
- **Request/apply path** (`ControllerApiConstants`, `UpdateStoreQueryParams`,
  `StoreConfigUpdater` for both `applyOnChild` and `applyOnParent`,
  `AdminExecutionTask`): plumbs the values from the admin message through to the
  persisted store.
- **admin-tool CLI**: new optional flags `--throughput-quota-in-bytes` (`tqib`)
  and `--throughput-quota-in-records` (`tqir`).

Trade-offs: this PR only makes the controller the SOT for the value; it does not
change quota *enforcement* (still external). A follow-up can project the
controller value to the enforcement layer.

###  Code changes
- [ ] Added new code behind **a config**. If so list the config names and their default values in the PR description.
- [ ] Introduced new **log lines**.
  - [ ] Confirmed if logs need to be **rate limited** to avoid excessive logging.

###  **Concurrency-Specific Checks**
Both reviewer and PR author to verify
- [x] Code has **no race conditions** or **thread safety issues**. (No new concurrency introduced; follows existing single-writer admin-consumption path.)
- [x] Proper **synchronization mechanisms** (e.g., `synchronized`, `RWLock`) are used where needed.
- [x] No **blocking calls** inside critical sections that could lead to deadlocks or performance degradation.
- [x] Verified **thread-safe collections** are used (e.g., `ConcurrentHashMap`, `CopyOnWriteArrayList`).
- [x] Validated proper exception handling in multi-threaded code to avoid silent thread termination.

## How was this PR tested?

- [x] New unit tests added.
  - `UpdateStoreQueryParamsTest`: round-trip (including the `-1` sentinel) + unset-is-empty edge case.
  - `TestZKStore`: default `-1`, setter/getter, and `cloneStore()` preservation.
  - `TestAdminTool`: CLI arg parsing for value, `-1`, and unset cases.
- [x] New integration tests added.
  - `TestAdminToolEndToEnd.testUpdateStoreThroughputQuotaViaAdminTool`: a fully **unmocked**
    prod-path test. It runs the actual admin-tool CLI entrypoint (`AdminTool.main` with
    `--throughput-quota-in-bytes` / `--throughput-quota-in-records`) against a live two-layer
    multi-region cluster, then verifies the quota persists through the full
    `AdminTool -> ControllerClient -> parent controller -> admin channel -> child controller -> ZooKeeper -> getStore`
    chain, asserting on **both** the parent (source of truth) and the child (propagated) regions.
    Passing locally (44.5s).
- [x] Modified or extended existing tests.
  - `AdminExecutionTaskTest`: non-breaking helper overload + a message->params propagation assertion.
- [x] Verified backward compatibility (if applicable).
  - Reuses existing schema fields; default `-1` preserves current behavior; no protocol version bump.

## Does this PR introduce any user-facing or breaking changes?
- [ ] No. You can skip the rest of this section.
- [x] Yes. Clearly explain the behavior change and its impact.

Adds two **optional** store configs (`throughputQuotaInBytes`,
`throughputQuotaInRecords`) settable via new admin-tool flags
(`--throughput-quota-in-bytes`, `--throughput-quota-in-records`). Both default to
`-1` (no limit), so existing stores are unaffected and no behavior changes unless
an operator explicitly sets a value. This is additive, not breaking.

### CODE REVIEW

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/storeconfig/StoreConfigUpdater.java [hunk 1]
```diff
@@ -289,6 +291,8 @@ public static void applyOnChild(
     Optional<Long> maxCompactionLagSeconds = params.getMaxCompactionLagSeconds();
     Optional<Integer> maxRecordSizeBytes = params.getMaxRecordSizeBytes();
     Optional<Integer> maxNearlineRecordSizeBytes = params.getMaxNearlineRecordSizeBytes();
+    Optional<Long> throughputQuotaInBytes = params.getThroughputQuotaInBytes();
+    Optional<Long> throughputQuotaInRecords = params.getThroughputQuotaInRecords();
     Optional<Boolean> unusedSchemaDeletionEnabled = params.getUnusedSchemaDeletionEnabled();
```
[the remark this answers is not in the document]
Comment (misyel) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/storeconfig/StoreConfigUpdater.java:310]: Good catch — fixed in b44cd4a54a. Added a shared `validateThroughputQuota` helper in `StoreConfigUpdater` that rejects any value `< -1` (only `-1` is the "no limit" sentinel), and wired it into **both** `applyOnChild` and `applyOnParent` so an invalid quota never reaches the store metadata or the UPDATE_STORE admin message. Covered by three new `StoreConfigUpdaterTest` cases: child reject `<-1`, child `-1` boundary allowed, and parent reject `<-1` (all passing).

File: internal/venice-common/src/main/java/com/linkedin/venice/meta/StoreInfo.java [hunk 2]
```diff
@@ -942,6 +948,22 @@ public void setMaxNearlineRecordSizeBytes(int maxNearlineRecordSizeBytes) {
...
+  }
+
+  public long getThroughputQuotaInRecords() {
+    return this.throughputQuotaInRecords;
+  }
+
+  public void setThroughputQuotaInRecords(long throughputQuotaInRecords) {
+    this.throughputQuotaInRecords = throughputQuotaInRecords;
+  }
+
```
Comment (kvargha) [inline internal/venice-common/src/main/java/com/linkedin/venice/meta/StoreInfo.java:966]: We should add some validation that it's a hybrid store

File: internal/venice-common/src/main/java/com/linkedin/venice/meta/StoreInfo.java [hunk 3]
```diff
@@ -942,6 +948,22 @@ public void setMaxNearlineRecordSizeBytes(int maxNearlineRecordSizeBytes) {
...
+    this.throughputQuotaInBytes = throughputQuotaInBytes;
+  }
+
+  public long getThroughputQuotaInRecords() {
+    return this.throughputQuotaInRecords;
+  }
+
+  public void setThroughputQuotaInRecords(long throughputQuotaInRecords) {
+    this.throughputQuotaInRecords = throughputQuotaInRecords;
+  }
```
Comment (kvargha) [inline internal/venice-common/src/main/java/com/linkedin/venice/meta/StoreInfo.java:965]: How is the internal system for enforcement going to be wired in here?

**Review by kvargha — COMMENTED** [review]
Please remove internal references from the PR description

File: internal/venice-common/src/main/java/com/linkedin/venice/meta/StoreInfo.java (hunk 3 above)
Comment (misyel) [reply internal/venice-common/src/main/java/com/linkedin/venice/meta/StoreInfo.java:965]: In a followup pr, we will introduce a store update callback interface that will run after the store metadata is updated and the callback will handle updates to the internal system. The callback will run before the admin execution is checkpointed so if it fails, the updates to the internal system will automatically be retried

File: internal/venice-common/src/main/java/com/linkedin/venice/meta/StoreInfo.java (hunk 2 above)
Comment (misyel) [reply internal/venice-common/src/main/java/com/linkedin/venice/meta/StoreInfo.java:966]: I took a look at the gating for checking if it's a hybrid store before setting the field and I think it is better to keep it un-gated because:

- other hybrid-only fields like maxNearlineRecordSizeBytes and nearlineProducerCompressionEnabled don't have any hybrid check before setting the field and adding the validation for this field will introduce strict batch -> hybrid ordering before the field can be set
- setting the field on a batch only store won't have any effect on quota enforcement so it should be safe to set it without hybrid enabled

What do you think?

Comment (kvargha) [reply internal/venice-common/src/main/java/com/linkedin/venice/meta/StoreInfo.java:966]: Fair

**Review by kvargha — APPROVED** [review]
